### PR TITLE
Added support for HiFi build for ISS (simulator) under EXECUTORCH_BUILD_CADENCE cmake-config switch

### PR DESCRIPTION
**Pull request by cad-audio**
…LD_CADENCE cmake-config switch.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -443,7 +443,9 @@ target_link_options_shared_lib(executorch)
 # Real integrations should supply their own YAML file that only lists the
 # operators necessary for the models that will run.
 #
+if(NOT EXECUTORCH_BUILD_CADENCE)
```
Comment (tarun292) [inline CMakeLists.txt:446]: Rather than guarding this by `NOT EXECUTORCH_BUILD_CADENCE` flags i think we should maybe add a flag to not build portable ops. e.g. `BUILD_EXECUTORCH_PORTABLE_OPS` which is on by default and for cadence builds it can be disabled explicitly.

File: CMakeLists.txt [hunk 2]
```diff
@@ -417,7 +417,7 @@ target_link_libraries(executorch_no_prim_ops PRIVATE program_schema)
 # Check if dl exists for this toolchain and only then link it.
 find_library(DL_LIBRARY_EXISTS NAMES dl)
 # Check if the library was found
-if(DL_LIBRARY_EXISTS)
+if(DL_LIBRARY_EXISTS AND NOT EXECUTORCH_BUILD_CADENCE)
```
Comment (tarun292) [inline CMakeLists.txt:420]: Can you please elaborate on why this change was needed?

Comment (larryliu0820) [reply CMakeLists.txt:420]: Would love this to not bound to any specific backend though. Maybe we can have an option `EXECUTORCH_USE_DL` and you guys can set it to False.

Comment (cad-audio) [reply CMakeLists.txt:420]: I agree, that would be better.

File: CMakeLists.txt (hunk 1 above)
Comment (cad-audio) [reply CMakeLists.txt]: updated the PR as per this suggestion

File: CMakeLists.txt (hunk 2 above)
Comment (cad-audio) [reply CMakeLists.txt]: updated the PR as per this suggestion

File: CMakeLists.txt [hunk 3]
```diff
@@ -201,6 +201,22 @@ cmake_dependent_option(
   "NOT EXECUTORCH_BUILD_ARM_BAREMETAL" OFF
 )
 
+#
+# portable_ops_lib: build portable_ops library. Disable, if not required
+#
+cmake_dependent_option(
+  BUILD_EXECUTORCH_PORTABLE_OPS "Build portable_ops library." ON
+  "NOT EXECUTORCH_BUILD_CADENCE" OFF
```
Comment (tarun292) [inline CMakeLists.txt:209]: I don't think we need to tie these together right. I think in the build commands you have for Cadence you can just explicitly turn BUILD_EXECUTORCH_PORTABLE_OPS off? Same with EXECUTORCH_USE_DL.

Comment (cad-audio) [reply CMakeLists.txt]: Done

**Comment by tarun292** [comment]
@cad-audio thanks for addressing the comments.

**Review by tarun292 — APPROVED** [review]

**Comment by mcremon-meta** [comment]
@cad-audio still looking into it, but maybe a rebase would help to see if CI gets happier
